src/community/flatgeobuf/src/main/java/org/geoserver/wfs/flatgeobuf/FlatGeobufOutputFormat.java

```diff
@@ -0,0 +1,85 @@
+/* (c) 2014 - 2016 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfs.flatgeobuf;
+
+import org.wololo.flatgeobuf.geotools.FeatureCollectionConversions;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import net.sf.json.JSONException;
+import org.geoserver.config.GeoServer;
+import org.geoserver.platform.Operation;
+import org.geoserver.platform.ServiceException;
+import org.geoserver.wfs.WFSGetFeatureOutputFormat;
+import org.geoserver.wfs.WFSInfo;
+import org.geoserver.wfs.request.FeatureCollectionResponse;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.feature.FeatureCollection;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+/**
+ * A GetFeatureInfo response handler specialized in producing FlatGeobuf data for a GetFeatureInfo
+ * request.
+ *
+ * @author Björn Harrtell
+ */
+public class FlatGeobufOutputFormat extends WFSGetFeatureOutputFormat {
+    private final Logger LOGGER = org.geotools.util.logging.Logging.getLogger(this.getClass());
+
+    public FlatGeobufOutputFormat(GeoServer gs) {
+        super(gs, "application/flatgeobuf");
+    }
+
+    /** capabilities output format string. */
+    public String getCapabilitiesElementName() {
+        return "FlatGeobuf";
+    }
+
+    /** Returns the mime type */
+    public String getMimeType(Object value, Operation operation) throws ServiceException {
+        return "application/flatgeobuf";
```

Comment:
Should it be "application/x-flatgeobuf"? Or is it already registered at IANA? Just thinking out loud here, not a problem to keep it as is.

Reply:
Need to involved OCG (or another similar org.) to get application/flatgeobuf, so yes perhaps it should be application/x-flatgeobuf for now.